### ISSUE TITLE
how to distrust dockerhub?

### ISSUE THREAD

**Issue by ghostsquad** [hadolint/hadolint#446]
- [] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

Is there a way to distrust dockerhub? Such that I can verify images are only ever pulled from my locally maintained registry or ECR or GCR?

**Comment by ghostsquad** [comment]
Similar to #401

**Comment by lorenzo** [comment]
I'll look into this for the next release

**Comment by logicbomb-1** [comment]
@lorenzo This is highly needed. Would be great if that can comes as a part for next release. Can you update when we can get to see it?:)

**Comment by maxbraun** [comment]
Maybe I'm holding it wrong, please enlighten me in case. I assume one has a particular, trusted and exclusive registry. Linting must highlight in case of no registry is set and / or the registry is not configured to be trusted. 

Here is my test setup:

Version: `Haskell Dockerfile Linter 1.19.0-no-git` installed via brew.

Dockerfile:
```dockerfile
# registry has to be set, fail
FROM alpine:edge as default-registry
# registry set, but not trusted, fail
FROM gcr.io/somewhere/alpine:edge as untrusted-registry
# registry set, trusted via cli, ok
FROM registry.example/alpine:edge as trusted-registry
```
.hadolint.yaml
```yaml
trustedRegistries:
  - registry.example

```
Execution:
```bash
$ hadolint Dockerfile
Dockerfile:2 DL3026 Use only an allowed registry in the FROM image
Dockerfile:4 DL3026 Use only an allowed registry in the FROM image
```
So hadolint complains about alpine:edge, which is implicitly dockerhub, the gcr.io which is not truested, but does not complain on registry.example.

Isn't that the behavior we'd like to have?

**Comment by lorenzo** [comment]
It would be nice to have. If anyone wants to implement it, please let me know

**Comment by maxbraun** [comment]
I think I was not precise enough: it was working like this in my test.

**Comment by lorenzo** [comment]
Ah, sorry. I misread
